pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/ThresholdShedder.java

```diff
@@ -233,5 +237,23 @@ private Pair<Boolean, String> getMaxUsageBroker(
         }
         return Pair.of(hasBrokerBelowLowerBound, maxUsageBrokerName);
     }
+    @Override
+    public void onBrokerChange(Set<String> newBrokers) {
+        synchronized (activeBrokers) {
+            activeBrokers.clear();
+            activeBrokers.addAll(newBrokers);
+        }
```

Comment:
Why not update the `brokerAvgResourceUsage` without maintaining a new `activeBrokers` map?

Reply:
If we modify `brokerAvgResourceUsage` directly, the entire `ThresholdShedder` needs to be locked every time, because there are 2 threads modify `brokerAvgResourceUsage` at the same time.
And now we don't need to lock every time, only when broker changes